db/blob/blob_file_builder.cc

```diff
@@ -178,11 +179,13 @@ Status BlobFileBuilder::OpenBlobFileIfNeeded() {
   file->SetWriteLifeTimeHint(write_hint_);
 
   Statistics* const statistics = immutable_cf_options_->statistics;
-
+  bool should_checksum_handoff = ShouldChecksumHandoff(
```

Comment:
It seems like many close copies of this code are being added in many places.
* Is there a reason this one does not set handoff_checksum_type in FileOptions?
* Can we share this code, perhaps in something called `NewDBWritableFileWriter`? Is the problem DBOptions vs. ImmutableCFOptions, etc.?

Reply:
I'm a little confuse, there is no FileType or handoff_checksum_type information in the FileOptions. So we may not be able to use the information in the FileOptios.

You mean, we pass the FileType as the parameter of WritableFileWriter constructor, and process the logic inside?

Reply:
Example of code in other places that is not present here:

https://github.com/facebook/rocksdb/blob/484e1ffade99fae516057d03a4dfef418a54770a/db/builder.cc#L138-L140

Reply:
> Example of code in other places that is not present here:
> 
> https://github.com/facebook/rocksdb/blob/484e1ffade99fae516057d03a4dfef418a54770a/db/builder.cc#L138-L140

Oh, I think I miss understood your previous comments. Will add the handoff_checksum_type here. Also, let me double check the code. If possible, I will remove "should_checksum_handoff" and replaced with the FileType as the input, such that ShouldChecksumHandoff is part of the construction function of WritableFileWriter.